lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```
@@ -117,6 +119,113 @@ sub list {
     $self->render(json => {JobTemplates => \@templates});
 }
 
+sub validate_yaml {
+    my ($self, $yaml, $validate_schema) = shift;
```

`shift` should likely be `@_`.

---

this can't have worked - so how is this covered? :)

---

Turns out it failed silently in the most beautiful way. And I didn't have tests for validation errors - with those the whole things looks much better now.